.github/workflows/sonar.yaml

```
@@ -0,0 +1,91 @@
+name: Sonar Scan
+
+on:
+  workflow_dispatch: # Trigger manually
+  pull_request:
+    paths-ignore:
+      - '**/*.md'
+      - '**/*.txt'
+      - 'charts/**'
+      - '.config/**'
+      - 'docs/**'
+      - '!docs/src/api/**'
+      - 'local/**'
+      - 'CHANGELOG.md'
+
+  push:
+    branches:
+      - main
+    tags:
+      - '**'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - name: Build with Gradle
+        working-directory: poc
+        run: ./gradlew build
+
+  check_sonar_configured:
+    runs-on: ubuntu-latest
+    steps:
+      - name: check_sonar_configured
+        run: |
+          echo "Checking if sonar is configured: ${{ env.SONAR_CONFIGURED }}"
+    env:
+      SONAR_CONFIGURED: ${{ secrets.SONAR_TOKEN != '' && secrets.SONAR_PROJECT_KEY != '' && secrets.SONAR_ORGANIZATION != '' }}
+    outputs:
+      sonar_configured: ${{ env.SONAR_CONFIGURED }}
+
+  analyze_with_Sonar:
+    needs: [check_sonar_configured]
+    # No need to run if we cannot use the sonar token
+    if: >-
+      needs.check_sonar_configured.outputs.sonar_configured == 'true'
+      && (github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == github.repository) 
+      && github.actor != 'dependabot[bot]'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of sonar analysis
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      #      - name: Cache maven packages
+      #        uses: actions/cache@v4
+      #        with:
+      #          path: ~/.m2
+      #          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+      #          restore-keys: ${{ runner.os }}-m2
```

Remove if not needed. Maybe use the built-in Gradle cache of the setup-java action
```suggestion
      - name: Set up JDK 17
        uses: actions/setup-java@v4
        with:
          java-version: '17'
          distribution: temurin
          cache: gradle
```